src/test/java/de/zalando/aruha/nakadi/validation/JSONSchemaValidationTest.java

```diff
@@ -76,15 +76,19 @@ public void schemaValidationShouldRespectEventTypeDefinition() {
         final EventTypeValidator validator = EventValidation.lookup(et);
 
         final JSONObject validEvent = new JSONObject(
+                "{\"foo\": \"bar\", \"bar\": {\"foo\": \"baz\"}, \"extra\": \"i should be no problem\", \"name\": \"12345\"}");
+        final JSONObject invalidEventMissingBar = new JSONObject(
                 "{\"foo\": \"bar\", \"extra\": \"i should be no problem\", \"name\": \"12345\"}");
-        final JSONObject invalidEventMissingFoo = new JSONObject(
-                "{\"extra\": \"i should be no problem\", \"name\": \"12345\"}");
         final JSONObject invalidEventMissingNameField = new JSONObject(
-                "{\"foo\": \"bar\", \"extra\": \"i should be no problem\"}");
+                "{\"foo\": \"bar\", \"bar\": {\"foo\": \"baz\"}, \"extra\": \"i should be no problem\"}");
+        final JSONObject invalidEventMissingMultipleFields = new JSONObject(
+                "{\"bar\": {\"foobar\": \"baz\"}, \"extra\": \"i should be no problem\", \"name\": \"12345\"}");
 
         assertThat(validator.validate(validEvent), isAbsent());
-        assertThat(validator.validate(invalidEventMissingFoo).get().getMessage(), equalTo("#: required key [foo] not found"));
+        assertThat(validator.validate(invalidEventMissingBar).get().getMessage(), equalTo("#: required key [bar] not found"));
         assertThat(validator.validate(invalidEventMissingNameField).get().getMessage(), equalTo("name is required"));
+        assertThat(validator.validate(invalidEventMissingMultipleFields).get().getMessage(),
+                equalTo("#: 2 schema violations found\n#/bar: required key [foo] not found\n#: required key [foo] not found"));
```

Comment:
Nested errors are displayed like this. First it displays the path, like "/bar/foo/baz/whatever" than the error message. So there's no need for recursively looking into causing exceptions, from what I got so far.